node/core/pvf/common/src/worker.rs

```diff
@@ -101,6 +101,11 @@ pub fn worker_event_loop<F, Fut>(
 		}
 	}
 
+	// Delete env vars to prevent malicious code from accessing them.
+	for (key, _) in std::env::vars() {
+		std::env::remove_var(key);
+	}
```

Comment:
We should just not inherit any env variables when starting the worker: https://doc.rust-lang.org/std/process/struct.Command.html#method.env_clear

(But we should still forward `RUST_LOG`)

Reply:
Nice, I didn't think of `RUST_LOG`. There's still an env var present with this method:

```
Jun 04 16:34:24.259  WARN parachain::pvf-prepare-worker: Unexpected env var found. key="__CF_USER_TEXT_ENCODING"
```

Not a blocker, just weird. Your method is still better.

Reply:
Hmm I don't know what that env var is but maybe we should still clear it on the child-side as it can be a source of randomness for attackers. 🤷‍♂️

Reply:
`__CF_USER_TEXT_ENCODING` this is some macos env variable, not sure why it still appears..